# Algortihm Parameters
alpha: 0.1
gamma: 0.8
epsilon: 0.9
epsilon_discount: 0.999 # 1098 eps to reach 0.1
nepisodes: 100000
nsteps: 1000

# Environment Parameters
desired_pose:
    x: 0.0
    y: 0.0
    z: 1.0
desired_force: 7.08 # In Newtons, normal contact force when stanting still with 9.81 gravity
desired_yaw: 0.0 # Desired yaw in radians for the hopper to stay
max_height: 3.0   # in meters
min_height: 0.5   # in meters
max_incl: 1.6       # in rads
running_step: 0.0010   # in seconds
joint_increment_value: 0.05   # in radians
done_reward: -1000.0 # reward
alive_reward: 100.0 # reward

list_of_observations: # Observations you want to use for the learning
  #- distance_from_desired_point
  - base_roll
  - base_pitch
  - base_yaw
  #- base_angular_vel_x
  #- base_angular_vel_y
  #- base_angular_vel_z
  #- base_linear_acceleration_x
  #- base_linear_acceleration_y
  #- base_linear_acceleration_z
  - contact_force
  #- joint_states_haa
  #- joint_states_hfe
  #- joint_states_kfe
  #- joint_effort_haa
  #- joint_effort_hfe
  #- joint_effort_kfe

discrete_division: 50 # Parts in which we divide each of the observations sensory range

weight_r1: 0.0 # Weight for joint positions ( joints in the zero is perfect )
weight_r2: 0.0 # Weight for joint efforts ( no efforts is perfect )
weight_r3: 0.0 # Weight for contact force similar to desired ( weight of monoped )
weight_r4: 0.0 # Weight for orientation ( vertical is perfect )
weight_r5: 0.0 # Weight for distance from desired point ( on the point is perfect )

init_joint_pose:
  haa: 0.0
  hfe: 0.0
  kfe: -1.6

joint_limits_array:
  haa_max: 1.6
  haa_min: -1.6
  hfe_max: 1.6
  hfe_min: -1.6
  kfe_max: 0.0
  kfe_min: -1.6

episode_done_criteria:
  - monoped_minimum_height
  #- monoped_vertical_orientation

maximum_base_linear_acceleration: 3000.0 # Based on making fall from a body height vertical max/min imu readings

maximum_base_angular_velocity: 20.0 # Based on free fall, one body length, at 45 degree angle max/min imu readings 20.0

maximum_joint_effort: 10.0 # Based on observation of Falling knee flexed

jump_increment: 0.8 # Increase in Radians for the jump motion. The bigger the higher the jump.